### PR TITLE
Fix service ARN parsing for ELB source (#524)

### LINKED ISSUE #524: ELB ARN is not built correctly when multi-level S3 key prefix is used to store access logs
**Describe what happened:**
ARN is not built correctly in `parse_service_arn` function when multi-level S3 key prefix (multiple `/` used) is used for storing access logs

**Describe what you expected:**
The ARN for ELBs to be built correctly irrespective of the S3 key prefix.

**Steps to reproduce the issue:**
AWS Account ID: `123456789123`
AWS Region: `us-east-1`
ALB name: `my-alb-name`
S3 key prefix for access logs: `elasticloadbalancing/my-alb-name`

execute the code block for `source == "elb"` from `parsing.parse_service_arn` for a key like:
`elasticloadbalancing/my-alb-name/AWSLogs/123456789123/elasticloadbalancing/us-east-1/2022/02/08/123456789123_elasticloadbalancing_us-east-1_app.my-alb-name.123456789aabcdef_20220208T1127Z_10.0.0.2_1abcdef2.log.gz`
the result will be `arn:aws:elasticloadbalancing:us-east-1:my-alb-name:loadbalancer/app/my-alb-name/123456789aabcdef` instead of `arn:aws:elasticloadbalancing:us-east-1:123456789123:loadbalancer/app/my-alb-name/123456789aabcdef`

### PR DESCRIPTION
**Pull request by vtstanescu**
### What does this PR do?

Closes #524 

### Types of changes

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [x] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)
- [x] This PR passes the unit tests 
- [ ] This PR passes the installation tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: aws/logs_monitoring/parsing.py [hunk 1]
```diff
@@ -365,11 +365,16 @@ def parse_service_arn(source, key, bucket, context):
         # 3. We extract the id of the loadbalancer
         # 4. We build the arn
         idsplit = key.split("/")
-        # If there is a prefix on the S3 bucket, idsplit[1] will be "AWSLogs"
-        # Remove the prefix before splitting they key
-        if len(idsplit) > 1 and idsplit[1] == "AWSLogs":
-            idsplit = idsplit[1:]
-            keysplit = "/".join(idsplit).split("_")
+        if not len(idsplit):
```
Comment (hghotra) [inline aws/logs_monitoring/parsing.py]: Can this be simplified to:
```suggestion
        if not idsplit:
```

Comment (hghotra) [reply aws/logs_monitoring/parsing.py]: Let's also take this opportunity to add a couple of debug lines:

```
if not idsplit:
    logger.debug("Invalid service ARN, unable to parse ELB ARN")
    return

# If there is a prefix on the S3 bucket, remove the prefix before splitting the key
if idsplit[0] != "AWSLogs":
    try:
        idsplit = idsplit[idsplit.index("AWSLogs"):]
        keysplit = "/".join(idsplit).split("_")
    except ValueError:
        logger.debug("Invalid S3 key, doesn't contain AWSLogs")
        return
```

Comment (vtstanescu) [reply aws/logs_monitoring/parsing.py]: implemented suggested changes

Comment (vtstanescu) [reply aws/logs_monitoring/parsing.py]: @hghotra please approve the workflows again, made a fix according to linter requirements

**Comment by vtstanescu** [comment]
maintainers please review & approve the workflows

**Comment by vtstanescu** [comment]
@tianchu @hghotra Thanks for reviewing and approving. I see the pipeline passed. Can this get merged and what is the release schedule?

**Pull request merged**